cdap-common/src/main/java/co/cask/cdap/common/conf/ConfigurationUtil.java

```diff
@@ -94,35 +91,35 @@ public static Configuration setAll(@Nullable Map<String, String> map, Configurat
 
 
   /**
-   * Prefixes the specified configurations with the name, and sets them onto the job's configuration.
+   * Prefixes the specified configurations with the given prefix, and sets them onto the job's configuration.
    *
    * @param conf the Configuration object on which the configurations will be set on
-   * @param name the name to use to prefix the keys of the configuration
+   * @param confKeyPrefix the String to prefix the keys of the configuration
    * @param namedConf the configuration values to be set
    */
-  public static void setNamedConfigurations(Configuration conf, String name, Map<String, String> namedConf) {
-    String confKeyPrefix = PREFIXED_CONF_PREFIX + name + CONF + ".";
+  public static void setNamedConfigurations(Configuration conf, String confKeyPrefix, Map<String, String> namedConf) {
     for (Map.Entry<String, String> entry : namedConf.entrySet()) {
       conf.set(confKeyPrefix + entry.getKey(), entry.getValue());
     }
   }
 
   /**
-   * Retrieves all configurations that are prefixed with a particular name.
+   * Retrieves all configurations that are prefixed with a particular prefix.
    *
    * @see {@link #setNamedConfigurations(Configuration, String, Map)}.
    *
    * @param conf the Configuration from which to get the configurations
-   * @param name the prefix to
+   * @param confKeyPrefix the prefix to search for in the keys
    * @return a map of key-value pairs, representing the requested configurations, after removing the prefix
    */
-  public static Map<String, String> getNamedConfigurations(Configuration conf, String name) {
+  public static Map<String, String> getNamedConfigurations(Configuration conf, String confKeyPrefix) {
     Map<String, String> namedConf = new HashMap<>();
-
-    String confKeyPrefix = PREFIXED_CONF_PREFIX + name + CONF;
-    Map<String, String> properties = conf.getValByRegex(confKeyPrefix + "\\..*");
+    int prefixLength = confKeyPrefix.length();
+    // since its a regex match, we want to look for the character '.', and not match any character
+    confKeyPrefix = confKeyPrefix.replace(".", "\\.");
+    Map<String, String> properties = conf.getValByRegex("^" + confKeyPrefix + ".*");
```

Comment:
can you add a unit test to ConfigurationUtilTest that would fail without this change? Or adapt the existing test....